src/components/Caption/Caption.tsx

```diff
@@ -0,0 +1,82 @@
+import { h, Component } from "preact";
+import * as styles from "./Caption.scss";
+import { secontsToTime } from "../../utils";
+
+type CaptionProps = {
+    caption: any;
+    seekTo(time: number): void;
+    highlighted: boolean;
+    isAutoScrollEnabled: boolean;
```

Comment:
this logic is not part of the caption, it should be handled in the `transcript`. Obviously only the component should have access to the internal ref. To make it happen you should expose a method named `scrollTo()` and call it from `transcript` when needed